### ISSUE TITLE
empty language and urldate fields

### ISSUE THREAD

**Issue by hartib** [plk/biblatex#528]
With the recent update there is an issue with empty entry fields now. I'm not sure if this behavior is caused by the updated biber 2.7 (MiKTeX 2.9) or the updated biblatex package (3.7).

Entry types with an empty language field cause a TeX error:
```
! Missing \endcsname inserted.
<to be read again>
                   \begingroup
\ifcsdef #1->\ifcsname #1
                         \endcsname \expandafter \@firstoftwo \else \expanda...

\blx@clearlang@i #1->\ifcsdef {blx@rlm@#1}
                                           {\expandafter \expandafter \expan...

\blx@citeprint ...k@citekey@next}\blx@beglangcite
                                                  \blx@execute \blx@loopcode...

\etb@forlistloop@i ...&->\ifblank {#2} {} {#1{#2}}
                                                  \ifblank {#3} {\listbreak ...

\blx@citei@cite ...{postnote}#4}\blx@citeloop {#3}
                                                  \endgroup \mkbibbrackets ...lx@setsfcodes \bibopenbracket #1
                                                  \bibclosebracket \endgroup \blx@cite@cite ...lxciteicmd {cite}{#1}{#2}{#3}{}}
                                                  #4\endgroup <to be read again>
```

An empty urldate appears in the TeX output as ```(visited on )```. It should probably rather create no text.

**Comment by plk** [comment]
Can you give a MWE for this issue?

**Comment by hartib** [comment]
Sure. Bib and Tex files are attached. Under windows we use the following scripts:

compile.bat:
---------------
@echo off

rem Manual and examples see %ProgramFiles(x86)%\MiKTeX 2.9\doc\latex\biblatex\

set mainfile=mwe
echo compiling "%mainfile%"...
call :compilefunc "%mainfile%"

goto end

:compilefunc
del *.aux
del "%~1".bbl
del "%~1".bcf
del "%~1".blg
del "%~1".cut
del "%~1".lof
del "%~1".log
del "%~1".lot
del "%~1".out
del "%~1".pdf
del "%~1".toc
del "%~1".xml
del "%~1".run.xml
del comment.cut

xelatex "%~1"
biber "%~1"
xelatex "%~1"
xelatex "%~1"
goto:eof


rem With LaTeX/BibTeX, this file is processed as follows:

rem latex  file

rem bibtex file1-blx

rem bibtex file2-blx

rem bibtex file3-blx

rem latex  file
rem Note that \usepackage[backend=biber]{biblatex} needs to be changed

:end

clean.bat:
------------
@echo off

set mainfile=mwe
echo cleaning "%mainfile%"...
call :cleanfunc "%mainfile%"

goto end

:cleanfunc
del *.aux
del "%~1".bbl
del "%~1".bcf
del "%~1".blg
del "%~1".cut
del "%~1".lof
del "%~1".log
del "%~1".lot
del "%~1".out
del "%~1".toc
del "%~1".xml
del "%~1".run.xml
del comment.cut
goto:eof

:end

------------------------------------------------

**Comment by plk** [comment]
I don't think your attachments work on github - can you  just paste the most basic self-contained file which demonstrates the issue?

**Comment by hartib** [comment]
I edited the comment already. It now contains everything.

**Comment by plk** [comment]
This was a bug, fixed in biber 2.8 DEV version now on Sourceforge
